kmod/src/xattr.c

```diff
@@ -620,10 +686,10 @@ static int scoutfs_xattr_set(struct dentry *dentry, const char *name,
 	if (unknown_prefix(name))
 		return -EOPNOTSUPP;
 
-	if (scoutfs_xattr_parse_tags(name, name_len, &tgs) != 0)
+	if (scoutfs_xattr_parse_tags(name, name_len, &tgs, sb) != 0)
 		return -EINVAL;
 
-	if ((tgs.hide | tgs.srch | tgs.totl) && !capable(CAP_SYS_ADMIN))
+	if ((tgs.hide | tgs.srch | tgs.totl | tgs.worm) && !capable(CAP_SYS_ADMIN))
 		return -EPERM;
 
```

Comment:
Let's add another test: EINVAL if tgs.worm is set but tgs.hide is *not* set.   That forces .worm. to be used with .hide. so that it's always private to the scoutfs archiving layer.   If it wasn't hidden it would be exposed to archiving utilities that try to save and restore xattrs, and they could try and set worm expiration policies as part of restore which is probably not right.